plugin-gradle/src/main/java/com/diffplug/gradle/spotless/FormatExtension.java

```diff
@@ -925,7 +925,8 @@ protected void setupTask(SpotlessTask task) {
 			steps.replaceAll(formatterStep -> formatterStep.filterByContent(OnMatch.EXCLUDE, targetExcludeContentPattern));
 		}
 		task.setSteps(steps);
-		task.setLineEndingsPolicy(getLineEndings().createPolicy(getProject().getProjectDir(), () -> totalTarget));
+		File projectDir = getProject().getProjectDir();
```

Comment:
Is there a way we can use the projectLayout API here to resolve this directory lazily? https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:troubleshooting

Reply:
The problem is the directory is immediately passed to [LineEnding.createPolicy()](https://github.com/diffplug/spotless/blob/main/lib/src/main/java/com/diffplug/spotless/LineEnding.java#L51) which is part of a Gradle-agnostic library.

Reply:
Ah, sounds good then